Prim and kruskal

JackJessada opened the pull request.

JackJessada committed:
3:20

JackJessada committed:
Create prim.py

JackJessada commented:
Prim's algorithm

JackJessada committed:
remove unnecessary file

monshinawatra commented on prim.py:
```diff
@@ -0,0 +1,20 @@
+counter=0
+differ = {} #ไว้เช็คว่าตัวไหนน้อยที่สุด, find minimum in dictionary
+check_if_it_have =  [] #check what vertex have been gone
+while counter<len(all_vertex)-1:
```
Based on the information provided. It looks like you forgot to declare the `all_vertex` variable.

monshinawatra commented on prim.py:
```diff
@@ -0,0 +1,20 @@
+differ = {} #ไว้เช็คว่าตัวไหนน้อยที่สุด, find minimum in dictionary
+check_if_it_have =  [] #check what vertex have been gone
+while counter<len(all_vertex)-1:
+    checklist=graph.vertices2name((graph.get_adjacent_vertices(first))) #ตัวที่Aพุ่งไปหาได้ ['B', 'C'] , check what vertex A can go
```
I know that `graph` is `Undirected Graph` but you forgot to declare the variable. I recommend declaring this on top of the file to avoid any potential issues or errors.

monshinawatra commented on prim.py:
```diff
@@ -0,0 +1,20 @@
+    different_checklist = set(checklist)-set(check_if_it_have) #checkตัวที่ยังไม่เคยไป ['B', 'C'], check what vertex isn't go 
+    check_if_it_have.append(first) #add for check what vertex have been gone
+    for adj in (list(different_checklist)):
+        edge = (graph.get_edge_to(first, adj))
```
I have noticed that the code works as intended, but I'm unsure why you have included so many unnecessary parentheses in the code. In my opinion, removing these parentheses would improve the readability of the code and make it easier to understand. Is there a specific reason for including these parentheses, or could they be removed for the sake of clarity?

monshinawatra commented on prim.py:
```diff
@@ -0,0 +1,20 @@
+    first = min_vertex[1] #the first str is change into adjectcent str
+    counter += 1
+
+print(f"anwser is {ans} and total weight is {sum(ans.values())}")
```
Misspelled. "anwser" should write as "answer".

monshinawatra reviewed (changes requested):
Overall, thanks to Prim's algorithm for finding the minimum spanning tree, I am looking forward to seeing the implementation of Kruskal's algorithm as well, as mentioned in the pull request name. I hope that it will be included in a future update. Thank you for your hard work and dedication to improving this project.

JackJessada committed:
Update prim.py

Change the procedural command into function and remove some unnecessary
bracket, correct misspelling

JackJessada committed:
Update return mode

Update a specific return only answer

JackJessada committed:
check

Is the same old one is it?

JackJessada committed:
fix bug
